File: comparative_tests/go_tests/main/test_programming_basics.go
package main

import "fmt"

func TestIf(value int) {
	if value > 0 {
		fmt.Println(fmt.Sprintf("%d is positive", value))
	} else if value < 0 {
		fmt.Println(fmt.Sprintf("%d is negative", value))
	} else {
		fmt.Println(fmt.Sprintf("%d is zero", value))
	}
}

func TestSwitch(value int) {
	switch {
	case value > 0:
		fmt.Println(fmt.Sprintf("Value %d is positive", value))
	case value < 0:
		fmt.Println(fmt.Sprintf("Value %d is negative", value))
	default:
		fmt.Println(fmt.Sprintf("Value %d is zero", value))
	}
}

func TestFor() {
	fmt.Print("Counting from 1 to 10 using a for loop :\n")
	for i := 1; i <= 10; i++ {
		fmt.Println(i)
	}
}

func TestWhile() {
	fmt.Print("Counting from 1 to 10 using a while loop :\n")
	i := 1
	for i <= 10 {
		fmt.Println(i)
		i++
	}
}

func TestList() {
	fmt.Println("Creating a list of 5 elements :")
	list := []int{1, 2, 3, 4, 5}
	fmt.Println(list)

	fmt.Println("Adding an element to the list :")
	list = append(list, 6)
	fmt.Println(list)

	fmt.Println("Removing the last element of the list :")
	list = list[:len(list)-1]
	fmt.Println(list)
}

func TestMap() {
	fmt.Println("Creating a map of 3 elements :")
	m := map[string]int{
		"one":   1,
		"two":   2,
		"three": 3,
	}
	fmt.Println(m)

	fmt.Println("Adding an element to the map :")
	m["four"] = 4
	fmt.Println(m)

	fmt.Println("Removing an element from the map :")
	delete(m, "two")
	fmt.Println(m)
}

func main() {
	// Tests conditional statements
	fmt.Println("\nTest if statement:")
	TestIf(5)
	TestIf(-5)
	TestIf(0)

	fmt.Println("\nTest switch statement:")
	TestSwitch(5)
	TestSwitch(-5)
	TestSwitch(0)

	// Test iteration statements
	fmt.Println("\nTest for statement:")
	TestFor()

	fmt.Println("\nTest while statement:")
	TestWhile()

	// Test collections
	fmt.Println("\nTest list:")
	TestList()

	fmt.Println("\nTest map:")
	TestMap()

	// Test functions
	fmt.Println("\nTest functions:")
	fmt.Println("\nEvery test is in a function in this file.")
}

File: comparative_tests/python_tests/main/test_programming_basics.py
def test_if(value):
    if value > 0:
        print(f"{value} is positive")
    elif value < 0:
        print(f"{value} is negative")
    else:
        print(f"{value} is zero")

def test_switch(value):
    if value > 0:
        print(f"Value {value} is positive")
    elif value < 0:
        print(f"Value {value} is negative")
    else:
        print(f"Value {value} is zero")

def test_for():
    print("Counting from 1 to 10 using a for loop:")
    for i in range(1, 11):
        print(i)

def test_while():
    print("Counting from 1 to 10 using a while loop:")
    i = 1
    while i <= 10:
        print(i)
        i += 1

def test_list():
    print("Creating a list of 5 elements:")
    lst = [1, 2, 3, 4, 5]
    print(lst)

    print("Adding an element to the list:")
    lst.append(6)
    print(lst)

    print("Removing the last element of the list:")
    lst.pop()
    print(lst)

def test_dict():
    print("Creating a dictionary of 3 elements:")
    d = {
        "one": 1,
        "two": 2,
        "three": 3
    }
    print(d)

    print("Adding an element to the dictionary:")
    d["four"] = 4
    print(d)

    print("Removing an element from the dictionary:")
    del d["two"]
    print(d)

def main():
    # Tests conditional statements
    print("\nTest if statement:")
    test_if(5)
    test_if(-5)
    test_if(0)

    print("\nTest switch statement:")
    test_switch(5)
    test_switch(-5)
    test_switch(0)

    # Test iteration statements
    print("\nTest for statement:")
    test_for()

    print("\nTest while statement:")
    test_while()

    # Test collections
    print("\nTest list:")
    test_list()

    print("\nTest dictionary:")
    test_dict()

    # Test functions
    print("\nTest functions:")
    print("\nEvery test is in a function in this file.")

if __name__ == "__main__":
    main()
